metacatalog/api/add.py

```diff
@@ -159,9 +165,20 @@ def add_variable(session: 'Session', name: str, symbol: str, column_names: List[
         unit = api.find_unit(session=session,name=unit, return_iterator=True).first()
     else:
         raise AttributeError('unit has to be of type integer or string.')
-
+    
     attrs['unit_id'] = unit.id
 
+    # get the keyword
+    if keyword:
+        if isinstance(keyword, int):
+            keyword = api.find_keyword(session=session, id=keyword, return_iterator=True).one()
+        elif isinstance(keyword, str):
+            keyword = api.find_keyword(session=session, full_path=keyword, return_iterator=True).first()
+        else:
+            raise AttributeError('keyword has to be of type integer or string.')

```

Comment:
Would it be too much work to add a unittest here?

Reply:
I added a smal unittest which adds a unit and a variable and finds it afterwards.